### DISCUSSION TITLE
Question regarding spatial matching

### DISCUSSION THREAD

**Discussion by Ph03n1xdust** [colmap/colmap#2303] [Q&A]
Hi!

As far as i know, the spatial matching module uses the camera positions from the DB, and the DB is supposed to contain similar information as the images.txt does.

Yet it seems to me that the spatial matching uses the tx, ty and tz directly from the DB to calculate the distances, even though they are in the camera coordinate system (as they should be in the images.txt) leading to completely wrong matches when the cameras are facing different directions.

Am I missing something?

**Comment by jared-krauss** [comment]
Ever find an answer? Have a similar problem.

**Reply by s-rnreic** [reply]
do i get the question right? you would like to know the transformation on how to get from Camera coordinates to World coordinates? 
I get the idea from your question that you are confused by the direction your cameras point at? If i am correct with this assumption then i recommend looking at the transformation given in the documentation ($$ -R^T * t $$)
